src/coreclr/jit/lowerxarch.cpp

```diff
@@ -119,14 +121,16 @@ void Lowering::LowerStoreIndir(GenTreeStoreInd* node)
 //
 // Notes:
 //    Performs containment checks on the replacement node if one is created
-GenTree* Lowering::TryLowerMulToLshSubOrLshAdd(GenTreeOp* node)
+GenTree* Lowering::TryLowerMulWithConstant(GenTreeOp* node)
 {
     assert(node->OperIs(GT_MUL));
 
-// We do not do this optimization in X86 as it is not recommended.
-#if TARGET_X86
-    return nullptr;
-#endif // TARGET_X86
+    // Do not do this optimization with min-opts enabled as
+    // this could create more tmp locals that need to be optimized
```

Comment:
Do you mind sharing the before and after dumps and a way to repro this? I can take a look why the 2 temps are not related to each other even though they are referring to the same local var.

Reply:
What we talked about yesterday was about using `GT_LEA` where the index and base are the same `GT_LCL_VAR`. I've decided that constructing a new `GT_LEA` op isn't worth doing in lowering; that optimization is now purely done in codegen - so I don't think we have to worry about it.

Reply:
I think it's fair to remove this comment as it not accurate enough, and I'll remove this description from the PR itself too.